### PR TITLE
RPRBLND-1971: Create Transform node dependent of object from scene

### PR DESCRIPTION
**Pull request by DagerD**
### PURPOSE
Should be looks like Instance node: select blender object and get transformation from that object and assign it to USD stage

### EFFECT OF CHANGE
- Changed UI for Transform node.
- Added new node "Transform by Empty object".

### TECHNICAL STEPS
- UI improvements for Transform node
- Reworked compute method for Transform node
- New node TrasformByEmptyObject - it gets transformation from choosen Empty object from scene.

### CODE REVIEW

File: src/hdusd/usd_nodes/nodes/transformations.py [hunk 1]
```diff
@@ -101,17 +109,100 @@ def compute(self, **kwargs):
...
+                                                       prim.GetPath())
 
-        if self.toggle_scale:
-            usd_geom.AddScaleOp()
-            root_prim.GetAttribute('xformOp:scale').Set(self.scale[:3])
+        if obj:
+            usd_geom = UsdGeom.Xform.Get(stage, root_xform.GetPath())
+            usd_geom.AddTransformOp()
+            obj_matrix = obj.matrix_world.transposed()
+            matrix = Gf.Matrix4d(obj_matrix[0][0], obj_matrix[0][1], obj_matrix[0][2], 0,
```
Comment (VascoPi) [inline src/hdusd/usd_nodes/nodes/transformations.py:202]: Code improvements matrix = Gf.Matrix4d(obj.matrix_world.transposed()).

**Review by VascoPi — CHANGES_REQUESTED** [review]
Some details were discussed offline.

File: src/hdusd/usd_nodes/nodes/transformations.py [hunk 2]
```diff
@@ -37,49 +57,44 @@ def update_data(self, context):
         update=update_data
     )
 
-    toggle_translation: bpy.props.BoolProperty(update=update_data)
-    translation: bpy.props.FloatVectorProperty(update=update_data, subtype='TRANSLATION')
+    translation_x: bpy.props.FloatProperty(update=update_data, subtype='DISTANCE')
```
Comment (VascoPi) [inline src/hdusd/usd_nodes/nodes/transformations.py:60]: Code improvements:
```
 
    translation: bpy.props.FloatVectorProperty(update=update_data, unit='LENGTH')
    rotation: bpy.props.FloatVectorProperty(update=update_data, unit='ROTATION')
    scale: bpy.props.FloatVectorProperty(update=update_data, unit='NONE', default=(1.0, 1.0, 1.0))


    def draw_buttons(self, context, layout):
        col = layout.column()
        col.prop(self, 'name')
        col.separator()
        col.row().prop(self, 'translation',  text='Translation')
        col.row().prop(self, 'rotation', text='Rotation')
        col.row().prop(self, 'scale', text='Scale')
```

File: src/hdusd/usd_nodes/nodes/transformations.py [hunk 3]
```diff
@@ -99,19 +114,91 @@ def compute(self, **kwargs):
...
+            override_prim = stage.OverridePrim(root_xform.GetPath().AppendChild(prim.GetName()))
+            override_prim.GetReferences().AddReference(input_stage.GetRootLayer().realPath,
+                                                       prim.GetPath())
 
-        if self.toggle_scale:
-            usd_geom.AddScaleOp()
-            root_prim.GetAttribute('xformOp:scale').Set(self.scale[:3])
+        if obj:
+            UsdGeom.Xform.Get(stage, root_xform.GetPath()).AddTransformOp()
+            root_prim.GetAttribute('xformOp:transform').Set(Gf.Matrix4d(obj.matrix_world.transposed()))
```
Comment (bnagirniak) [inline src/hdusd/usd_nodes/nodes/transformations.py:202]: use get_transform() from export/object.py

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Some improvements are required

**Review by VascoPi — APPROVED** [review]
Works well.

File: src/hdusd/usd_nodes/nodes/transformations.py [hunk 4]
```diff
@@ -99,19 +165,17 @@ def compute(self, **kwargs):
...
-            root_prim.GetAttribute('xformOp:scale').Set(self.scale[:3])
+    def depsgraph_update(self, depsgraph):
+        if not self.object:
+            return
 
-        return stage
+        obj = next((update.id for update in depsgraph.updates if isinstance(update.id, bpy.types.Object)
+                    and not update.id.hdusd.is_usd and update.id.name == self.object.name), None)
+        if obj:
+            self.reset()
```
Comment (bnagirniak) [inline src/hdusd/usd_nodes/nodes/transformations.py]: add new line

**Review by bnagirniak — APPROVED** [review]
Tested - works good

**Pull request merged**